### ISSUE TITLE
[Bug] Task execution failed

### ISSUE THREAD

**Issue by gelinhua** [nunchaku-tech/nunchaku#788]
### Checklist

- [x] 1. I have searched for related issues and FAQs (https://nunchaku.tech/docs/nunchaku/faq/faq.html) but was unable to find a solution.
- [x] 2. The issue persists in the latest version.
- [ ] 3. Please note that without environment information and a minimal reproducible example, it will be difficult for us to reproduce and address the issue, which may delay our response.
- [ ] 4. If your report is a question rather than a bug, please submit it as a discussion at https://github.com/mit-han-lab/nunchaku/discussions/new/choose. Otherwise, this issue will be closed.
- [ ] 5. If this is related to ComfyUI, please report it at https://github.com/mit-han-lab/ComfyUI-nunchaku/issues.
- [ ] 6. I will do my best to describe the issue in English.

### Describe the Bug

When I edit the image through the command line, it gets stuck at a step without any error. I tried about 10 times and this problem occurred twice.
<img width="2292" height="578" alt="Image" src="https://github.com/user-attachments/assets/2e76789a-2dd2-4f38-b312-ddbb64a0fa02" />

### Environment

GPU: 5070TI, RAM: 64G，nunchaku1.0.1

---python pip list------
 uv pip list
Package            Version
------------------ --------------
accelerate         1.11.0
anyio              4.11.0
certifi            2025.10.5
charset-normalizer 3.4.4
click              8.3.0
colorama           0.4.6
diffusers          0.36.0.dev0
einops             0.8.1
filelock           3.20.0
fsspec             2025.9.0
h11                0.16.0
hf-xet             1.2.0
httpcore           1.0.9
httpx              0.28.1
huggingface-hub    0.34.3
idna               3.11
importlib-metadata 8.7.0
jinja2             3.1.6
markupsafe         2.1.5
mpmath             1.3.0
networkx           3.5
numpy              2.3.4
nunchaku           1.0.1+torch2.8
packaging          25.0
peft               0.17.1
pillow             12.0.0
protobuf           6.33.0
psutil             7.1.1
pyside6            6.8.3
pyside6-addons     6.8.3
pyside6-essentials 6.8.3
pyyaml             6.0.3
regex              2025.10.23
requests           2.32.5
safetensors        0.6.2
sentencepiece      0.2.1
setuptools         70.2.0
shellingham        1.5.4
shiboken6          6.8.3
sniffio            1.3.1
sympy              1.14.0
tokenizers         0.22.1
torch              2.8.0+cu128
torchvision        0.23.0+cu128
tqdm               4.67.1
transformers       4.57.1
typer-slim         0.20.0
typing-extensions  4.15.0
urllib3            2.5.0
zipp               3.23.0

### Reproduction Steps

python imageEdit.py

---code---
import torch
from diffusers import QwenImageEditPlusPipeline
from diffusers.utils import load_image

from nunchaku import NunchakuQwenImageTransformer2DModel
from nunchaku.utils import get_gpu_memory, get_precision

rank = 128  # you can also use rank=128 model to improve the quality

# Load the model
transformer = NunchakuQwenImageTransformer2DModel.from_pretrained(
    f"C:/Users/Admin/codes/models/nunchaku-tech/nunchaku-qwen-image-edit-2509/svdq-{get_precision()}_r{rank}-qwen-image-edit-2509.safetensors"
)

pipeline = QwenImageEditPlusPipeline.from_pretrained(
    "C:/Users/Admin/codes/models/Qwen/Qwen-Image-Edit-2509", transformer=transformer, torch_dtype=torch.bfloat16
)

if get_gpu_memory() > 18:
    pipeline.enable_model_cpu_offload()
else:
    # use per-layer offloading for low VRAM. This only requires 3-4GB of VRAM.
    transformer.set_offload(
        True, use_pin_memory=False, num_blocks_on_gpu=1
    )  # increase num_blocks_on_gpu if you have more VRAM
    pipeline._exclude_from_cpu_offload.append("transformer")
    pipeline.enable_sequential_cpu_offload()

image1 = load_image("C:/Users/Admin/codes/resources/images/image0.jpeg")
image1 = image1.convert("RGB")
#image2 = load_image("C:/Users/Admin/codes/resources/images/image5.jpeg")
#image2 = image2.convert("RGB")
#image3 = load_image("C:/Users/Admin/codes/resources/images/image2.jpeg")
#image3 = image3.convert("RGB")

def onStepEnd(pipe, step, t, callback_kwargs):
    print(f"Step {step} completed.")
    return callback_kwargs

#prompt = "let the woman in image1 stand on the grass"
#prompt = """clean image without watermark, intact scene, no text, high-resolution, sharp details, natural texture, seamless inpainting """
prompt = "干净无水印原图, 完整画面, 高分辨率, 细节清晰, 自然纹理, 无缝修复"

inputs = {
    "image": [image1],
    "prompt": prompt,
    "true_cfg_scale": 4.0,
    "negative_prompt": "", #"水印, logo, 字幕, 签名, 日期, 二维码, 马赛克, 遮挡, 裁剪边缘, 低质量, 模糊, 失真, 噪点, 过曝, 欠曝, 压缩块",
    "num_inference_steps": 30,
    "callback_on_step_end": onStepEnd,
}

output = pipeline(**inputs)
output_image = output.images[0]
output_image.save(f"qwen-image-edit-2509-r{rank}.png")
import time
import os
output_dir = "../output"
if not os.path.exists(output_dir):
    os.makedirs(output_dir)
output_image.save(f"{output_dir}/{time.strftime('%Y-%m-%d-%H-%M-%S')}-qwen-image-edit-r{rank}.png")

**Comment by gelinhua** [comment]
这个问题我来回答，经过验证是因为模型在子线程中运行，它们跨线程使用与 Qt 共享 GPU/GL 上下文时在非隔离环境中初始化/移动，导致 native 层（C++/CUDA/FFI）崩溃。PySide6与 PyTorch / Transformers / Diffusers 对 GPU / native runtime 的初始化时引发了底层访问冲突。通过使用主-子进程模式，把推理放入到子进程，解决了该问题，经过两天的使用，未发生类似情况。异常如下：
Current thread 0x0000343c (most recent call first): File "C:\Users\Admin\codes\.venv\Lib\site-packages\transformers\tokenization_utils_base.py", line 810 in to File "C:\Users\Admin\codes\.venv\Lib\site-packages\diffusers\pipelines\qwenimage\pipeline_qwenimage.py", line 204 in _get_qwen_prompt_embeds File "C:\Users\Admin\codes\.venv\Lib\site-packages\diffusers\pipelines\qwenimage\pipeline_qwenimage.py", line 254 in encode_prompt File "C:\Users\Admin\codes.venv\Lib\site-packages\diffusers\pipelines\qwenimage\pipeline_qwenimage.py", line 606 in __call__ File "C:\Users\Admin\codes\.venv\Lib\site-packages\torch\utils\_contextlib.py", line 120 in decorate_context File "C:\Users\Admin\codes\pipeline\pipelineManager.py", line 167 in genImage File "C:\Users\Admin\codes\pipeline\pipelineManager.py", line 137 in startGen Thread 0x00002494 (most recent call first): File "C:\Users\Admin\AppData\Roaming\uv\python\cpython-3.12.10-windows-x86_64-none\Lib\threading.py", line 359 in wait File "C:\Users\Admin\AppData\Roaming\uv\python\cpython-3.12.10-windows-x86_64-none\Lib\threading.py", line 655 in wait File "C:\Users\Admin\codes\.venv\Lib\site-packages\tqdm\_monitor.py", line 60 in run File "C:\Users\Admin\AppData\Roaming\uv\python\cpython-3.12.10-windows-x86_64-none\Lib\threading.py", line 1075 in _bootstrap_inner File "C:\Users\Admin\AppData\Roaming\uv\python\cpython-3.12.10-windows-x86_64-none\Lib\threading.py", line 1032 in _bootstrap Thread 0x0000753c (most recent call first): File "C:\Users\Admin\codes\main.py", line 121 in main File "C:\Users\Admin\code\amainpy", line 125 in <module>
